trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/plugins/generator/cacsl2boogietranslator/witness/GraphMLCorrectnessWitnessExtractor.java

```diff
@@ -105,35 +106,34 @@ protected ExtractedCorrectnessWitness extractWitness() {
 				continue;
 			}
 			worklist.addAll(current.getOutgoingNodes());
-			final Map<IASTNode, ExtractedWitnessInvariant> match = matchWitnessToAstNode(current, mStats);
+			final var match = matchWitnessToAstNode(current, mStats);
 			map = mergeMatchesIfNecessary(map, match);
 		}
-		final ExtractedCorrectnessWitness rtr = new ExtractedCorrectnessWitness();
-		rtr.addWitnessStatements(map);
 		mLogger.info("Processed " + closed.size() + " nodes");
-		return rtr;
+		return new GraphMLExtractedCorrectnessWitness(map);
 	}
 
-	protected Map<IASTNode, ExtractedWitnessInvariant> mergeMatchesIfNecessary(
-			final Map<IASTNode, ExtractedWitnessInvariant> mapA, final Map<IASTNode, ExtractedWitnessInvariant> mapB) {
+	protected Map<IASTNode, Pair<ImmutableSet<String>, ExtractedWitnessInvariant>> mergeMatchesIfNecessary(
+			final Map<IASTNode, Pair<ImmutableSet<String>, ExtractedWitnessInvariant>> mapA,
```

Comment:
Would it make sense to introduce a dedicated data type for `Pair<ImmutableSet<String>, ExtractedWitnessInvariant>` ?